src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+
+import React, { useState, useMemo, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar from "./TableFilterBar";
+
+interface Props {
+    data: { [key: string]: string | number | null }[];
+    headers: { [key: string]: string | null };
+}
+
+const Table: React.FC<Props> = (props) => {
+    const [filterText, setFilterText] = useState<{ [key: string]: string }>({});
+
+    // temporary solution to fix hydration error
+    const [domLoaded, setDomLoaded] = useState(false);
+    useEffect(() => {
+        setDomLoaded(true);
+    }, []);
+
+    const subHeaderComponentMemo = useMemo(() => {
+        const handleClear: () => void = () => {
+            if (filterText) {
+                setFilterText({});
+            }
+        };
+        return (
+            <TableFilterBar
+                onFilter={(e: React.ChangeEvent<HTMLInputElement>, filterField: string): void => {
+                    const filterTextCopy = { ...filterText };
```

Comment:
is this equivalent to `{ ...filtertext, filterField: e.target.value }` ?

Reply:
From my understanding, the two are not equivalent.

The current code: takes in a filterField as a string then sets the corresponding field in filterText to a value of e.target.value
eg. console.log(filterText["full_name"]) ---> Tom Jerry
 
The suggested code: sets the field call filterFied in filterText to a value of e.target.value
eg. console.log(filterText["filterField"]) ---> Tom Jerry

Reply:
That's completely correct!
Can this be solved with square brackets, like
`{ ...filtertext, [filterField]: e.target.value }`
?

Reply:
Thats a valid syntax. Thanks!